src/client/Microsoft.Identity.Client/MsalErrorMessage.cs

```diff
@@ -166,6 +166,7 @@ public static string iOSBrokerKeySaveFailed(string keyChainResult)
         public const string NonHttpsRedirectNotSupported = "Non-HTTPS URL redirect is not supported in webview. ";
 
         public const string IDTokenMustHaveTwoParts = "ID Token must contain at least 2 parts. ";
+        public const string IDTokenMustHaveThreeParts = "ID Token must contain at least 3 parts. ";
```

Comment:
maybe something like "IdToken must be of form JWT" which implies 3 parts, but is more specific. OIDC specifies the format as a JWT.

Reply:
We should not have ID token parsing logic in 2 different places and I have fixed that.

Reply:
I'll update the error message as well.

Reply:
Btw, IdToken actually is in format `<header>.<payload>.` i.e. missing sig.